src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -110,5 +108,59 @@ internal static bool CompletionRequiresQuotes(string completion, bool escape)
 
         private static bool ContainsCharsToCheck(ReadOnlySpan<char> text, bool escape)
             => text.ContainsAny(escape ? s_escapeCharsToCheck : s_defaultCharsToCheck);
+
+        /// <summary>
+        /// Quotes and escapes a given completion text based on the specified options.
+        /// </summary>
+        /// <param name="completionText">
+        /// The text to be quoted and potentially escaped.
+        /// </param>
+        /// <param name="quote">
+        /// The quote character to use for enclosing the text. Defaults to a single quote ('') if not provided.
+        /// </param>
+        /// <param name="escapeSingleQuoteChars">
+        /// Indicates whether single quote characters in the text should be escaped. Defaults to <c>true</c>.
+        /// </param>
+        /// <param name="escapeDoubleQuoteChars">
+        /// Indicates whether special characters (e.g., backticks and dollar signs) in the text
+        /// should be escaped when using double quotes. Defaults to <c>false</c>.
+        /// </param>
+        /// <param name="escapeGlobbingPathChars">
+        /// Indicates whether globbing path characters (e.g., square brackets) in the text should be escaped.
+        /// Defaults to <c>false</c>.
+        /// </param>
+        /// <returns>
+        /// The quoted and optionally escaped version of the <paramref name="completionText"/>.
+        /// </returns>
+        internal static string QuoteCompletionText(
+            string completionText,
+            string quote,
+            bool escapeSingleQuoteChars = true,
+            bool escapeDoubleQuoteChars = false,
+            bool escapeGlobbingPathChars = false)
```

Comment:
It is not used anywhere. Rather, it should be removed if there is no clear understanding that it should be used, i.e. to fix existing bugs. 
I don't have a clear understanding in which cases and which characters should be disabled. It would be nice to have a description of all such scenarios.

Reply:
Ok so I have removed `escapeDoubleQuoteChars` && `escapeGlobbingPathChars` since they are not used.